lib/logstash_json/event.ex

```diff
@@ -93,16 +93,24 @@ defmodule LogstashJson.Event do
     :binary.copy("0", count - byte_size(num)) <> num
   end
 
-  # Traverse complex objects and inspect PID's to their string representation
-  defp print_pids(it) when is_pid(it), do: inspect(it)
-  defp print_pids(it) when is_list(it), do: Enum.map(it, &print_pids/1)
-  defp print_pids(it) when is_tuple(it), do: print_pids(Tuple.to_list(it))
-  defp print_pids(%_{} = it), do: print_pids(Map.from_struct(it))
+  # traverse data and stringify special Elixir/Erlang terms
+  defp pre_encode(it) when is_pid(it), do: inspect(it)
+  defp pre_encode(it) when is_list(it), do: Enum.map(it, &pre_encode/1)
+  defp pre_encode(it) when is_tuple(it), do: pre_encode(Tuple.to_list(it))
+
+  defp pre_encode(%module{} = it) do
+    try do
+      :ok = Protocol.assert_impl!(Poison.Encoder, module)
```

Comment:
Went to check if there is a non-raising version of `assert_impl` but seems that there is not...